### PR TITLE
Better description of the import limiting for deserialization

### PR DESCRIPTION
**Pull request by potiuk**
Introduce better description and significant change in release notes for import limiting for deserialization.

Follow up after #27887

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by bolkedebruin** [comment]
The serializer will be generic and not just tied to XCom so referring to XCom doesn't make sense

**Comment by potiuk** [comment]
> The serializer will be generic and not just tied to XCom so referring to XCom doesn't make sense

But we can change the message then ? can we? For now it's only XCom, as I understand.

**Comment by potiuk** [comment]
> But we can change the message then ? can we? For now it's only XCom, as I understand.

Just some reasin why - It took me quite a while to look it up (going through Git History and finding PR) so I can only imagine how hard it might be for the user, and I think we should give as good a hint where it is from as we can.

File: airflow/utils/json.py [hunk 1]
```diff
@@ -212,7 +212,11 @@ def object_hook(self, dct: dict) -> object:
                     break
 
             if not cls:
-                raise ImportError(f"{classname} was not found in allow list for import")
+                raise ImportError(
+                    f"{classname} was not found in allow list for import in XCom. "
+                    "If you want to continue to use your class in XCom, add it to "
+                    "allowed_deserialization_classes config in core section of config."
```
Comment (kaxil) [inline airflow/utils/json.py:218]:
```suggestion
                    "allowed_deserialization_classes config in core section of airflow.cfg."
```

File: airflow/utils/json.py [hunk 2]
```diff
@@ -212,7 +212,11 @@ def object_hook(self, dct: dict) -> object:
                     break
 
             if not cls:
-                raise ImportError(f"{classname} was not found in allow list for import")
+                raise ImportError(
+                    f"{classname} was not found in allow list for import in XCom. "
```
Comment (kaxil) [inline airflow/utils/json.py:216]:
```suggestion
                    f"{classname} was not found in allow list for imports. "
```

**Comment by bolkedebruin** [comment]
As I intend to further refactor it in the earlier mentioned PR and in the intention is to have that out as part of 2.5.1 and this is just clarification I think I'd better solve it there.

**Comment by potiuk** [comment]
> As I intend to further refactor it in the earlier mentioned PR and in the intention is to have that out as part of 2.5.1 and this is just clarification I think I'd better solve it there.

Cool. Closing it then :)

**Pull request closed without merging**